apis/v1alpha2/tlsroute_types.go

```diff
@@ -49,26 +49,35 @@ type TLSRouteSpec struct {
 	CommonRouteSpec `json:",inline"`
 
 	// Hostnames defines a set of SNI names that should match against the
-	// SNI attribute of TLS ClientHello message in TLS handshake.
-	//
-	// SNI can be "precise" which is a domain name without the terminating
-	// dot of a network host (e.g. "foo.example.com") or "wildcard", which is
-	// a domain name prefixed with a single wildcard label (e.g. `*.example.com`).
-	// The wildcard character `*` must appear by itself as the first DNS label
-	// and matches only a single label. You cannot have a wildcard label by
-	// itself (e.g. Host == `*`).
-	//
-	// Requests will be matched against the SNI attribute in the following
-	// order:
-	//
-	// 1. If SNI is precise, the request matches this Route if the SNI in
-	//    ClientHello is equal to one of the defined SNIs.
-	// 2. If SNI is a wildcard, then the request matches this Route if the
-	//    SNI is to equal to the suffix (removing the first label) of the
-	//    wildcard.
-	// 3. If SNIs are unspecified, all requests associated with the gateway TLS
-	//    listener will match. This can be used to define a default backend
-	//    for a TLS listener.
+	// SNI attribute of TLS ClientHello message in TLS handshake. This matches
+	// the RFC 1123 definition of a hostname with 2 notable exceptions:
+	//
+	// 1. IPs are not allowed.
```

Comment:
IPs are not allowed in SNI in the RFC

Reply:
Any relevant use cases for supporting IPs?
Can we start narrow and open up later?

Reply:
I think you're both agreeing that IPs should not be allowed, but John is also saying we probably don't even need to specify that here.

Reply:
This is copied from other places which abide exactly by RFC1123, which *does* say that it SHOULD be possible to enter an IP in dotted-decimal format as a hostname, so it's important in those cases.

I agree that since SNI explicitly disallows literal addresses in the hostname field, it's not actually needed here, but I think it doesn't hurt and makes the rules across the different types of resources in our API more similar.

In short, I think we should probably keep this.

Reply:
Kept in place since the starting point is RFC 1123, but added a reference to RFC 6066 for this bullet point to explain that IP addresses were not allowed in SNI names.